src/main/java/org/jetbrains/research/refactorinsight/utils/PsiUtils.java

```diff
@@ -0,0 +1,94 @@
+package org.jetbrains.research.refactorinsight.utils;
+
+import com.intellij.lang.java.JavaLanguage;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiParameter;
+import com.intellij.psi.PsiType;
+import com.intellij.psi.search.PsiElementProcessor;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+public class PsiUtils {
+  // Suppresses default constructor, ensuring non-instantiability.
+  private PsiUtils() {
+  }
+
+  @Nullable
+  public static PsiMethod findMethod(@NotNull PsiFile psiFile, @NotNull String qualifiedName) {
+    if (psiFile.getLanguage().is(JavaLanguage.INSTANCE)) {
+      return findMethodJava(psiFile, qualifiedName);
+    } else {
+      return null;
+    }
+  }
+
+  @Nullable
+  public static PsiMethod findMethodJava(@NotNull PsiFile psiFile, @NotNull String qualifiedName) {
```

Comment:
We search for a method with the same signature, right? I suggest creating a new method that calculates the method's signature, it will simplify the understanding of this method.